apps/datapuller/src/class.ts

```diff
@@ -0,0 +1,69 @@
+import { IClassItem, NewClassModel } from "@repo/common";
+import { ClassesAPI } from "@repo/sis-api/classes";
+
+import { Config } from "./config";
+import setup from "./shared";
+import mapClassToNewClass, { CombinedClass } from "./shared/classParser";
+import { fetchActiveTerms, fetchPaginatedData } from "./shared/utils";
+
+export async function updateClasses(config: Config) {
+  const log = config.log;
+  const classesAPI = new ClassesAPI();
+
+  log.info("Fetching Active Terms");
+  const activeTerms = await fetchActiveTerms(log, {
+    app_id: config.sis.TERM_APP_ID,
+    app_key: config.sis.TERM_APP_KEY,
+  });
+
+  log.info(activeTerms);
+
+  const classes = await fetchPaginatedData<IClassItem, CombinedClass>(
+    log,
+    classesAPI.v1,
+    activeTerms,
+    "getClassesUsingGet",
+    {
+      app_id: config.sis.CLASS_APP_ID,
+      app_key: config.sis.CLASS_APP_KEY,
+    },
+    (data) => data.apiResponse.response.classes || [],
+    mapClassToNewClass
+  );
+
+  log.info("Example Class:", classes[0]);
+
+  await NewClassModel.deleteMany({
+    "session.term.id": { $in: activeTerms },
+  });
+
+  // Insert classes in batches of 5000
+  const insertBatchSize = 5000;
```

Comment:
Just leaving this here, but we should probably do some testing on `hozer` to determine the final batch size. It all depends on amount of memory, etc.